controllers/basket.controllers.js

```diff
@@ -0,0 +1,143 @@
+
+    const coownerBaskets = await (await Participants.find({ user: req.user._id }).populate('basket'))
+    .map((doc) => {
+        console.log(new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate())
+        doc.basket.isHot = (new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate() < 7);
+        return doc.basket;        
+    });
+
+    const baskets = [...ownerBaskets, ...coownerBaskets];
+
+    res.status(200).json([...baskets])
+}
+
+async function getPublicBaskets(req, res, next) {
+    if(!req.user){
+        return next(new HttpError('A user is not logged in.', 401));
+    }
+
+    const ownerBaskets = await Basket.find({ ownerId: req.user._id, isPublic: true });
+
+    const coownerBaskets = await (await Participants.find({ user: req.user._id }).populate('basket'))
+    .map((doc) => {
+        console.log(new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate())
+        doc.basket.isHot = (new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate() < 7);
+        return doc.basket;        
+    }).filter(basket => basket.isPublic === true);
+
+    const baskets = [...ownerBaskets, ...coownerBaskets];
+
+    res.status(200).json([...baskets])
+}
+
+async function getPrivateBaskets(req, res, next) {
+    if(!req.user){
+        return next(new HttpError('A user is not logged in.', 401));
+    }
+
+    const ownerBaskets = await Basket.find({ ownerId: req.user._id, isPublic: false });
+
+    const coownerBaskets = await (await Participants.find({ user: req.user._id }).populate('basket'))
+    .map((doc) => {
+        console.log(new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate())
+        doc.basket.isHot = (new Date(Date.parse(doc.basket.expirationDate) - Date.now()).getDate() < 7);
+        return doc.basket;        
+    }).filter(basket => basket.isPublic === false);
+
+    const baskets = [...ownerBaskets, ...coownerBaskets];
+
+    res.status(200).json([...baskets])
+}
+
+
+async function createBasket(req, res, next) {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+        return next(new HttpError('Invalid or not all inputs passed.', 422));
+    }
+
+    if(!req.user){
+        return next(new HttpError('A user is not logged in.', 401));
+    }
+
+    let basket;
+
+    try{
+        basket = await Basket.create({
+            ownerId: req.user._id,
+            name: req.body.name,
+            description: req.body.description,
+            goal: req.body.goal,
+            value: req.body.value,
+            expirationDate: req.body.expirationDate,
+            isPublic: req.body.isPublic,
+            creationDate: Date.now(),
+            image: req.body.image
+        })
+    }
+    catch (error){
+        return next(new HttpError(`Error when creating a basket appeared. Message: ${error.message}`, 500));
+    }
+
+    res.status(201).json({ id: basket._id, message: "Successfully created a basket." });
+}
```

Comment:
move let basket into try {} block (use const), and  move res.status()... right after const basket (in try {} block too)

Reply:
Fixed